### PR TITLE
Add SWC support

### PR DESCRIPTION
**Pull request by leoyli-headsup**
### What I did

- Add `@swc/core` transpiler support so that the consumer project can run typescript via `@swc/register` require hook.
- Related to #190.

Please let me know if the PR is okay, not sure the benchmark things though.

---

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "typescript": "tsd",
     "typescript:jest": "jest",
     "typescript:esm": "node scripts/unit-typescript-esm.js",
+    "typescript:swc": "node scripts/unit-typescript-swc.js",
```
Comment (zekth) [inline package.json:13]: you need to add this command to `test` script to be handled by the ci. Otherwise lgtm

**Review by zekth — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (leoyli-headsup) [reply package.json:13]: oh good call, thanks!

**Comment by Fdawgs** [comment]
LGTM.
Like the Mii avatar btw, takes me back!

**Review by Fdawgs — APPROVED** [review]

**Review by zekth — APPROVED** [review]
lgtm

**Pull request merged**